test/e2e/helpers_test.go

```diff
@@ -226,6 +233,58 @@ func verifyGateway(ctx context.Context, t *testing.T, env environments.Environme
 	}, gatewayUpdateWaitTime, time.Second)
 }
 
+func deployGatewayWithTCPListener(ctx context.Context, t *testing.T, env environments.Environment) *gatewayv1alpha2.Gateway {
```

Comment:
We might want to consider putting this into a separate file where TCPListener/TCPRoute related functions are stored.

Reply:
What was your concern here? the `helpers_test.go` would become a file for everything?

Reply:
Yup, exactly that

Reply:
Just to add to that: instead of naming this `tcp_test.go` we can do something like: `helpers_tcp_test.go`

Reply:
I think splitting a `helper_gateways_test.go` for functions related to gateway APIs would be a better choice. WDYT?

Reply:
took the splitting of files above. It's such a coincidence that the two split files has both 294 lines :-)

Reply:
I like this idea 👍